core/src/test/java/de/danoeh/antennapod/core/feed/LocalFeedUpdaterTest.java

```diff
@@ -0,0 +1,258 @@
+import android.net.Uri;
+import android.webkit.MimeTypeMap;
+
+import androidx.documentfile.provider.DocumentFile;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.shadows.ShadowMediaMetadataRetriever;
+import org.robolectric.shadows.util.DataSource;
+
+import java.io.File;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import de.danoeh.antennapod.core.ApplicationCallbacks;
+import de.danoeh.antennapod.core.ClientConfig;
+import de.danoeh.antennapod.core.R;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.storage.PodDBAdapter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.robolectric.Shadows.shadowOf;
+
+/**
+ * Test local feeds handling in class LocalFeedUpdater.
+ */
+@RunWith(RobolectricTestRunner.class)
+public class LocalFeedUpdaterTest {
+
+    /**
+     * URL to locate the local feed media files on the external storage (SD card).
+     * The exact URL doesn't matter here as access to external storage is mocked
+     * (seems not to be supported by Robolectric).
+     */
+    private static final String FEED_URL =
+            "content://com.android.externalstorage.documents/tree/primary%3ADownload%2Flocal-feed";
+
+    private Context context;
+    private File localFeedDir1;
+    private File localFeedDir2;
+
+    @Before
+    public void setUp() {
+
+        // Initialize environment
+        context = InstrumentationRegistry.getInstrumentation().getContext();
+        UserPreferences.init(context);
+
+        // Initialize database
+        PodDBAdapter.init(context);
+        PodDBAdapter.deleteDatabase();
+        PodDBAdapter adapter = PodDBAdapter.getInstance();
+        adapter.open();
+        adapter.close();
+
+        // Emulate turned off gpodder.net support in SyncService
+        SharedPreferences spref = mock(SharedPreferences.class);
+        when(spref.getString(eq("prefGpodnetHostname"), anyString())).thenReturn("gpodder.net");
+        Application app = mock(Application.class);
+        when(app.getSharedPreferences(anyString(), anyInt())).thenReturn(spref);
+        ClientConfig.applicationCallbacks = mock(ApplicationCallbacks.class);
+        when(ClientConfig.applicationCallbacks.getApplicationInstance()).thenReturn(app);
+
+        localFeedDir1 = new File("sampledata/local-feed1");
```

Comment:
Would it be possible to use a test flavour of the `assets` folder? From my understanding, `sampledata` is used for layout rendering in Android Studio, not for unit tests.

Reply:
Thank you for the hint regarding `sampledata`, I wasn't aware of that. I will look for an alternative solution in the next days.

Reply:
I moved the local feed media files from the `sampledata` folder to the `test/assets` folder. For file access `context.getAssets()` is used. Anything else I can improve?